newapi/ooniapi/measurements.py

```diff
@@ -1062,45 +1514,103 @@ def get_aggregated():
       '200':
         description: Returns aggregated counters
     """
+    # TODO:
+    #  better split of large dimensions in output?
+    #  add limit and warn user
     log = current_app.logger
     param = request.args.get
-    axis_x = param("axis_x")
-    axis_y = param("axis_y")
-    category_code = param("category_code")
-    domain = param("domain")
-    input = param("input")
-    test_name = param("test_name")
-    probe_asn = param("probe_asn")
-    probe_cc = param("probe_cc")
-    since = param("since")
-    until = param("until")
-    format = param("format", "JSON")
+    try:
+        axis_x = param_lowercase_underscore("axis_x")
+        axis_y = param_lowercase_underscore("axis_y")
+        category_code = param_uppercase("category_code")
+        domain = param_domain("domain")
+        inp = param_url("input")
+        test_name = param_lowercase_underscore("test_name")
+        probe_asn = param_asn("probe_asn")
+        probe_cc = param_uppercase("probe_cc")
+        since = param("since")
+        if since:
+            since = parse_date(since)
+        until = param("until")
+        if until:
+            until = parse_date(until)
 
-    dimension_cnt = int(bool(axis_x)) + int(bool(axis_y))
+        resp_format = param("format", "JSON").upper()
+        assert resp_format in ("JSON", "CSV")
+    except Exception as e:
+        return jsonify({"v": 0, "error": str(e)})
 
-    cacheable = until and parse_date(until) < datetime.now() - timedelta(hours=72)
+    if current_app.config["USE_CLICKHOUSE"]:
+        r = _clickhouse_aggregation(
+            resp_format,
+            since,
+            until,
+            inp,
+            domain,
+            category_code,
+            probe_cc,
+            probe_asn,
+            test_name,
+            axis_x,
+            axis_y,
+        )
+        return r
+
+    r = _postgresql_aggregation(
+        resp_format,
+        since,
+        until,
+        inp,
+        domain,
+        category_code,
+        probe_cc,
+        probe_asn,
+        test_name,
+        axis_x,
+        axis_y,
+    )
+    return r
+
+
+def _postgresql_aggregation(
+    resp_format,
+    since,
+    until,
+    inp,
+    domain,
+    category_code,
+    probe_cc,
+    probe_asn,
+    test_name,
+    axis_x,
+    axis_y,
+):
+
+    dimension_cnt = int(bool(axis_x)) + int(bool(axis_y))
+    cacheable = until and until < datetime.now() - timedelta(hours=72)
 
     # Assemble query
     def coalsum(name):
         return sql.text("COALESCE(SUM({0}), 0) AS {0}".format(name))
```

Comment:
Maybe then `coalsum` can be deleted?

Reply:
No, this is the postgresql version of the query.

Reply:
Fixed